### PR TITLE
Refresh log + support for `before` and `after` refresh middleware opts

### PR DESCRIPTION
**Pull request by cichli**
Depends on clojure-emacs/cider-nrepl#233. Current UI looks like this:

screenshot 2015-07-19 18 40 50

Feedback welcome!

### CODE REVIEW

**Comment by bbatsov** [comment]
The code looks great, but this definitely deserves a changelog entry and some documentation in the README. We wouldn't want people to be totally unaware about such great functionality, right?

File: cider-interaction.el [hunk 1]
```diff
@@ -183,6 +183,27 @@ if the candidate is not namespace-qualified."
   :group 'cider
   :package-version '(cider . "0.9.0"))
 
+(defconst cider-refresh-log-buffer "*cider-refresh-log*")
+
+(defcustom cider-refresh-before-fn nil
+  "Function for `cider-refresh' to call before reloading.
```
Comment (Malabarba) [inline cider-interaction.el]: "Clojure function ..." I had to read this twice to understand it wasn't an emacs hook. :)

**Comment by bbatsov** [comment]
Forget my remark about the changelog. I'm way too tired today...

File: cider-interaction.el (hunk 1 above)
Comment (cichli) [inline cider-interaction.el]: :+1: good spot, updated.

**Comment by cichli** [comment]
> The code looks great, but this definitely deserves a changelog entry and some documentation in the README.

Looks like `cider-refresh` isn't mentioned anywhere in the README at all right now - not even the keybindings! Putting something together now.

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**